### PR TITLE
Reverts reverted toast icons

### LINKED ISSUE #79415: Nar'sie summon makes ghosts' screen black
Reporting client version: 514.1589

## Round ID:
[217833](https://scrubby.melonmesa.com/round/217833)

## Testmerges:
- [Changes how halloween lighting works slightly](https://www.github.com/tgstation/tgstation/pull/79340)
- [Extends Halloween](https://www.github.com/tgstation/tgstation/pull/79316)

## Reproduction:
Be an observer
Have Nar'sie be summoned
Your screen goes black and you can't see anything

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
During the notify ghosts refactor I was unsure why we were creating a custom overlay for each ghost alert when new_master did the same. Turns out - it resets icons, which is useful here. This also caused #79415.

The original notify ghosts was performing this transformation operation O(n), which I've made O(1)
## Why It's Good For The Game
Fixes #79415

Screenshot 2023-11-05 131336
## Changelog
:cl:
fix: The screen alert should no longer break ghost UI when it's huge
/:cl:

### CODE REVIEW

File: code/__HELPERS/icons.dm [hunk 1]
```diff
@@ -1527,3 +1527,25 @@ GLOBAL_LIST_EMPTY(transformation_animation_objects)
...
+
+/// Fikou's fix for making toast alerts look nice - resets offsets, transforms to fit
+/proc/get_small_overlay(atom/source)
+	var/mutable_appearance/alert_overlay = new(source)
+	alert_overlay.pixel_x = 0
+	alert_overlay.pixel_y = 0
+	var/icon/size_check = icon(source.icon, source.icon_state)
+	var/scale = 1
+	var/width = size_check.Width()
+	var/height = size_check.Height()
```
Comment (Ghommie) [inline code/__HELPERS/icons.dm:1539]:
```suggestion
	var/list/icon_dimensions = get_icon_dimensions(source.icon)
	var/width = icon_dimensions["width"]
	var/height = icon_dimensions["height"]
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Review by Ghommie — APPROVED** [review]

**Comment by Cadunkus** [comment]
Unsure but I think this may have broken the icon that tells servant golems who their master is. I don't know much about code at all though so, like I said I'm not sure.

**Comment by jlsnow301** [comment]
> Unsure but I think this may have broken the icon that tells servant golems who their master is. 

could you possibly submit an issue about this? i'll look into it

**Comment by Cadunkus** [comment]
> > Unsure but I think this may have broken the icon that tells servant golems who their master is.
> 
> could you possibly submit an issue about this? i'll look into it

I only just found out golems no longer have a servant icon when dicking around on a private server. I could but it wouldn't say anything but "servant golems don't get an icon anymore" and have no further details than that.

**Pull request merged**